pkg/manifests/configfile.go

```diff
@@ -427,52 +454,56 @@ func (cf *ConfigFile) execGenerators(ctx context.Context, generators []Generator
 // execContainerImageUpdaters executes all the image updates in the configuration file.
 // It will stop at the first error, in which case the returned error will be non-nil
 func (cf *ConfigFile) execContainerImageUpdaters(ctx context.Context,
-	workload resource.ID, container string, image, imageTag string) []ConfigFileCombinedExecResult {
+	workload resource.ID, container string, image, imageTag string, defaultTimeout time.Duration) []ConfigFileCombinedExecResult {
 	env := makeEnvFromResourceID(workload)
 	env = append(env,
 		"FLUX_CONTAINER="+container,
 		"FLUX_IMG="+image,
 		"FLUX_TAG="+imageTag,
 	)
-	commands := []string{}
+	commands := []Command{}
 	var updaters []Updater
 	if cf.CommandUpdated != nil {
 		updaters = cf.CommandUpdated.Updaters
 	}
 	for _, u := range updaters {
-		commands = append(commands, u.ContainerImage.Command)
+		commands = append(commands, u.ContainerImage)
 	}
-	return cf.execCommandsWithCombinedOutput(ctx, env, commands)
+	return cf.execCommandsWithCombinedOutput(ctx, env, commands, defaultTimeout)
 }
 
 // execPolicyUpdaters executes all the policy update commands given in
 // the configuration file. An empty policyValue means remove the
 // policy. It will stop at the first error, in which case the returned
 // error will be non-nil
 func (cf *ConfigFile) execPolicyUpdaters(ctx context.Context,
-	workload resource.ID, policyName, policyValue string) []ConfigFileCombinedExecResult {
+	workload resource.ID, policyName, policyValue string, defaultTimeout time.Duration) []ConfigFileCombinedExecResult {
 	env := makeEnvFromResourceID(workload)
 	env = append(env, "FLUX_POLICY="+policyName)
 	if policyValue != "" {
 		env = append(env, "FLUX_POLICY_VALUE="+policyValue)
 	}
-	commands := []string{}
+	commands := []Command{}
 	var updaters []Updater
 	if cf.CommandUpdated != nil {
 		updaters = cf.CommandUpdated.Updaters
 	}
 	for _, u := range updaters {
-		commands = append(commands, u.Policy.Command)
+		commands = append(commands, u.Policy)
 	}
-	return cf.execCommandsWithCombinedOutput(ctx, env, commands)
+	return cf.execCommandsWithCombinedOutput(ctx, env, commands, defaultTimeout)
 }
 
-func (cf *ConfigFile) execCommandsWithCombinedOutput(ctx context.Context, env []string, commands []string) []ConfigFileCombinedExecResult {
+func (cf *ConfigFile) execCommandsWithCombinedOutput(ctx context.Context, env []string, commands []Command, defaultTimeout time.Duration) []ConfigFileCombinedExecResult {
 	env = append(env, "PATH="+os.Getenv("PATH"))
 	result := []ConfigFileCombinedExecResult{}
 	for _, c := range commands {
 		stdOutAndErr := bytes.NewBuffer(nil)
-		err := cf.execCommand(ctx, env, stdOutAndErr, stdOutAndErr, c)
+		timeout := c.Timeout
+		if &c.Timeout == nil || timeout.Duration < time.Second {
```

Comment:
`if &c.Timeout == nil` does not make sense as a comparison. It says "is the _address_ of a field in a value `nil`", which will never be true. If you want the timeout value to be optional, make the type of the field a pointer.

Reply:
I think I fixed this?